src/test/java/com/revature/controllers/AddressControllerTest.java

```diff
@@ -0,0 +1,197 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.json.AutoConfigureJsonTesters;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+
+import com.revature.dtos.AddressRequest;
+import com.revature.exceptions.UserNotFoundException;
+import com.revature.models.Address;
+import com.revature.models.User;
+import com.revature.services.AddressService;
+import com.revature.services.UserService;
+
+@AutoConfigureJsonTesters
+@WebMvcTest(AddressController.class)
+class AddressControllerTest {
+	@Autowired
+	private MockMvc mvc;
+	
+	@Autowired
+	private JacksonTester<Address> jsonAddress;
+	
+	@Autowired
+	private JacksonTester<Set<Address>> jsonReviewList;
+	
+	@Autowired
+	private JacksonTester<AddressRequest> JsonAddressRequest;
+	
+	@MockBean
+	private AddressService aserv;
+	
+	@MockBean
+	private UserService userv;
+	
+	@InjectMocks
+	private AddressController controller;
+	private Logger log = LoggerFactory.getLogger(AddressControllerTest.class);
+	
+	private final String MAPPING = "/api/addresses";
+	private Address dummyAddress;
+	private User dummyUser;
+	
+	@BeforeEach
+	void setUp() {
+		this.dummyAddress = new Address(1, "6803 N.Navarro st", "apt 332", "Victoria", "77904", "Texas", null);
+		this.dummyUser = new User(1, "email@email.com", "abx", "Primary", "User" , "Customer", null, null, null);
+	}
+	
+	@AfterEach
+	void tearDown() {
+		this.dummyUser = null;
+		this.dummyAddress = null;
+	}
+	
+	@Test
+	void testGetUserAddress_Success() throws Exception{
+		Set<Address> addresses = new HashSet<Address>();
+		addresses.add(this.dummyAddress);
+		dummyUser.setAddresses(addresses);
+		
+		given(this.userv.findById(1)).willReturn(Optional.of(dummyUser));
```

Comment:
I prefer to use the `getId` methods of each dummy object, but this appears to work fine for now.